lib/ops/opsservice/monitoring.go

```diff
@@ -34,6 +35,65 @@ import (
 	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
 )
 
+// GetClusterMetrics returns basic CPU/RAM metrics for the specified cluster.
+func (o *Operator) GetClusterMetrics(ctx context.Context, req ops.ClusterMetricsRequest) (*ops.ClusterMetricsResponse, error) {
+	return GetClusterMetrics(ctx, o.cfg.Metrics, req)
+}
+
+// GetClusterMetrics retrieves all cluster metrics from the provided client.
+func GetClusterMetrics(ctx context.Context, metrics monitoring.Metrics, req ops.ClusterMetricsRequest) (*ops.ClusterMetricsResponse, error) {
+	err := req.CheckAndSetDefaults()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	totalCPUCores, err := metrics.GetTotalCPU(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	currentCPURate, err := metrics.GetCurrentCPURate(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	maxCPURate, err := metrics.GetMaxCPURate(ctx, req.Interval)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	historicCPURate, err := metrics.GetCPURate(ctx, time.Now().Add(-req.Interval), time.Now(), req.Step)
```

Comment:
`GetCPURate` returns results based on the last time interval (similar to `GetMaxMemoryRate`) but expects `start, stop` - is this distinction useful, or you could simply pass just the `interval` and `step` instead?

Reply:
We could but the Prometheus API allows retrieving metrics for arbitrary time intervals so I wanted to have this possibility exposed in our API too.